SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialStatics.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialStatics.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "GeneralProjectSettings.h"
+#include "SpatialConstants.h"
+#include "SpatialGDKSettings.h"
+#include "Utils/ActorGroupManager.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialStatics);
+
+bool USpatialStatics::SpatialNetworkingEnabled()
+{
+    return GetDefault<UGeneralProjectSettings>()->bSpatialNetworking;
+}
+
+bool USpatialStatics::SpatialOffloadingEnabled()
+{
+    return SpatialNetworkingEnabled() && GetDefault<USpatialGDKSettings>()->bEnableOffloading;
+}
+
+bool USpatialStatics::IsAuthoritativeWorkerType(const AActor* Actor)
+{
+    if (Actor == nullptr)
+    {
+        UE_LOG(LogSpatialStatics, Warning, TEXT("Actor was nullptr in USpatialStatics::IsAuthoritativeWorkerType"));
+        return false;
+    }
+
+    if (!SpatialNetworkingEnabled() || !SpatialOffloadingEnabled()) 
+    {
+        return Actor->HasAuthority();
+    }
+
+    UWorld* World = Actor->GetWorld();
+    if (!World)
+    {
+        UE_LOG(LogSpatialStatics, Warning, TEXT("World was nullptr in USpatialStatics::IsAuthoritativeWorkerType for actor: %s", *Actor->GetName()));
+        return false;
+    }
+
+    UActorGroupManager* ActorGroupManager = Cast<USpatialNetDriver>(World->GetNetDriver())->ActorGroupManager;
+    return ActorGroupManager->GetWorkerTypeForClass(Actor->GetClass()).Compare(World->GetGameInstance()->GetSpatialWorkerType()) == 0;
+}
+
+bool USpatialStatics::IsDefaultWorker(const UObject* WorldContextObject)
```

Comment:
When do we use this? it feels like it would be pretty risky to use this in any code once users are able to change the default server not to be UnrealWorker right?

Reply:
I imagine using this if I want to make sure a given component (or piece of logic) runs only on the default worker (such as making parts of a game mode execute only on the game server).